### PR TITLE
Add a way to type Context::$data

### PR DESCRIPTION
**Pull request by lyrixx**

### CODE REVIEW

File: src/Context.php [hunk 1]
```diff
@@ -25,7 +25,9 @@ public function __construct(
         $this->currentDirectory = $currentDirectory ?? PathHelper::getRoot();
     }
 
-    /** @param array<(int|string), mixed> $data */
+    /**
+     * @phpstan-param ContextData $data The input parameter accepts an array or an Object
```
Comment (lyrixx) [inline src/Context.php:29]: This is wrong, we could add more data and/or modify only what's needed

Comment (lyrixx) [reply src/Context.php]: Fixed

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -0,0 +1,21 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Function args\\\\args\\(\\) has parameter \\$word with no type specified\\.$#"
```
Comment (pyrech) [inline phpstan-baseline.neon:4]: Why is this argument typing not fixed directly?

File: phpstan-baseline.neon [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+			count: 1
+			path: src/Context.php
+
+		-
+			message: "#^Parameter \\#1 \\$data of class Castor\\\\Context constructor expects array\\{name\\: string, production\\: bool, foo\\?\\: string\\}, array\\<int\\|string, mixed\\> given\\.$#"
+			count: 1
+			path: src/Context.php
+
+		-
+			message: "#^Parameter \\#1 \\$nodes of method PhpParser\\\\NodeTraverser\\:\\:traverse\\(\\) expects array\\<PhpParser\\\\Node\\>, array\\<PhpParser\\\\Node\\\\Stmt\\>\\|null given\\.$#"
```
Comment (pyrech) [inline phpstan-baseline.neon:19]: Can't we solve that too?

File: phpstan-baseline.neon (hunk 2 above)
Comment (lyrixx) [reply phpstan-baseline.neon:4]: It's done on purpose :)

File: phpstan-baseline.neon (hunk 3 above)
Comment (lyrixx) [reply phpstan-baseline.neon:19]: I don't know how to 👍🏼

Comment (pyrech) [reply phpstan-baseline.neon]: Fixed

File: examples/args.php [hunk 4]
```diff
@@ -8,6 +8,9 @@
 
 use function Castor\run;
 
+/**
+ * @param string[] $argument2
+ */
 #[AsTask(description: 'Dumps all arguments and options, with custom configuration')]
 function args(
     #[AsArgument(description: 'This is a required argument', suggestedValues: ['hello', 'bonjour', 'hola'])]
```
Comment (pyrech) [inline examples/args.php:16]:
```suggestion
    #[AsArgument(description: 'This is a required argument without any typing', suggestedValues: ['hello', 'bonjour', 'hola'])]
```

**Review by pyrech — APPROVED** [review]

**Pull request merged**